pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -35,6 +35,29 @@
 
 warning_cache = WarningCache()
 
+BType = Union[torch.Tensor, str, Mapping[Any, 'BType'], Iterable['BType']]
+
+
+def extract_batch_size(batch: BType) -> int:
+    """
+    Recursively unpack a batch to find a torch.Tensor.
+
+    Returns:
+        ``len(tensor)`` when found, or ``1`` when it hits an empty or non iterable.
+    """
+    if isinstance(batch, torch.Tensor):
+        return batch.size(0)
+    if isinstance(batch, str):
+        return len(batch)
+    if isinstance(batch, dict):
+        sample = next(iter(batch.values()), 1)
+        return extract_batch_size(sample)
+    if isinstance(batch, Iterable):
+        sample = next(iter(batch), 1)
+        return extract_batch_size(sample)
+
+    return 1
+
```

Comment:
thanks for adding this and the test @kandluis ! Since this more general than the logger connector, could we move this to a separate utilities file? @carmocca what do you think?

Reply:
Happy to! Where would you recommend we land this? Would `utilities/data.py` be a good place?

Reply:
that seems like a reasonable spot

Reply:
Thanks for the help! Moved and added the `test_data.py` unit tests.